### PR TITLE
Update FH_auto_classes.py

### PR DESCRIPTION
**Pull request by gta4hater**
I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!

### CODE REVIEW

File: src/FH_auto_classes.py [hunk 1]
```diff
@@ -136,7 +136,11 @@ def analyzeInput(self) -> None:
 
         # adjust the allowed downshifting rpm range,
         # depending on the aggressiveness
-        self.rpm_range_bottom = max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size)
+        #(-400 for lower rpm cars and it looks if the rpm range is below or = 6100) else it defaults to normal self.rpm_range_bottom 
+        self.rpm_range_bottom = (
+        max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size) - 400 
+        if (self.rpm_range_top <= 6100) 
+        else max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size)
```
Comment (GinoLin980) [inline src/FH_auto_classes.py:143]: can be simplified for better readability

```python
        self.rpm_range_bottom = max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size)
        if (self.rpm_range_top <= 6100):
            self.rpm_range_bottom -= 400
```

**Comment by GinoLin980** [comment]
> I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!

@gta4hater 
Do you think this could be triggered by `kickdown` feature? I met that problem before but it was when developing `kickdown` feature. But I can't see anything causing it from the current code.

Will ignoring user input(gas, brake) for a few seconds(or an amount of data) be a good idea to prevent this situation.

And I think there's **latency between local network communication and processing latency**, which might result in this problem at hard braking.(maybe we're already from 2nd gear to 1st gear in game. but the program still receives 2nd gear due to the latency, therefore it performs down shift again?)

**Comment by GinoLin980** [comment]
Oh, you mean you have fixed this situation?

**Comment by gta4hater** [comment]
> > I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!
> 
> @gta4hater Do you think this could be triggered by `kickdown` feature? I met that problem before but it was when developing `kickdown` feature. But I can't see anything causing it from the current code.
> 
> Will ignoring user input(gas, brake) for a few seconds(or an amount of data) be a good idea to prevent this situation.
> 
> And I think there's **latency between local network communication and processing latency**, which might result in this problem at hard braking.(maybe we're already from 2nd gear to 1st gear in game. but the program still receives 2nd gear due to the latency, therefore it performs down shift again?)

i printed the gears and the latency was spot, it was instant as far i know (i printed the gear and throttle). The issue was in the downshift logic from my if statement. I changed it because after the change the issue where it shfited into reverse didnt happend. 

which makes sense because the other function didnt have the new variable for the lower rpm cars so the kickdown "function" had the old variable and made it shift one more gear down? well idk now it works lol

**Comment by gta4hater** [comment]
> Oh, you mean you have fixed this situation? Nevermind, I'll leave this thought here if anyone needs it in the future :)

yep

**Comment by GinoLin980** [comment]
> > Oh, you mean you have fixed this situation? Nevermind, I'll leave this thought here if anyone needs it in the future :)
> 
> yep 

I made a comment on your code. 
Please have a look. 
https://github.com/GinoLin980/Forza-Horizon-realistic-gearbox/pull/5#discussion_r1819938385

**Comment by gta4hater** [comment]
im not the best in python ngl :(((

**Comment by GinoLin980** [comment]
It's fine my brother. 
Imma merge this PR.

**Pull request merged**